CodesmithProjects/TrailBlazers
# Added session management and protected features behind active session

## Overview

Issue Type


Description
Session database created.
Logging in will now create a session in the database. Sessions will automatically expire after 15 minutes.
Features related to leaving reviews/favoriting trails will only be available to users with a valid session.

Ticket Item

FEATURE #4: Session management

Steps to Validate Feature

1. Try to save a trail as a favorite without being logged in
2. Go to pgAdmin and create a query for 'SELECT * FROM favorite_trails'
3. no new entries will show
4. Login with a valid account
5. Go to pgAdmin and create a query for 'SELECT * FROM sessions'
6. a new session will appear
7. Try to save a trail as a favorite
8. Go to pgAdmin and create a query for 'SELECT * FROM favorite_trails'
9. new trail will appear

Screenshots & Videos

![Screenshot 2023-04-14 at 5 01 39 PM](https://user-images.githubusercontent.com/118780080/232154367-8dbca290-d7ac-426e-988c-a6b385a4b1ba.png)
